cmd/copy.go

```diff
@@ -436,6 +438,68 @@ func (raw rawCopyCmdArgs) cookWithId(jobId common.JobID) (cookedCopyCmdArgs, err
 		cooked.includeAfter = &parsedIncludeAfter
 	}
 
+	versionsChan := make(chan string)
+	var filePtr *os.File
+	// Get file path from user which would contain list of all versionIDs
+	// Process the file line by line and then prepare a list of all version ids of the blob.
+	if raw.listOfVersionIDs != "" {
+		filePtr, err = os.Open(raw.listOfVersionIDs)
+
+		if err != nil {
+			return cooked, fmt.Errorf("cannot open %s file passed with the list-of-file flag", raw.listOfFilesToCopy)
+		}
+	}
+
+	go func() {
+		defer close(versionsChan)
+
+		addToChannel := func(v string, paramName string) {
+			// empty strings should be ignored, otherwise the source root itself is selected
+			if len(v) > 0 {
+				raw.warnIfHasWildcard(includeWarningOncer, paramName, v)
+				versionsChan <- v
+			}
+		}
+
+		if filePtr != nil {
+			scanner := bufio.NewScanner(filePtr)
+			checkBOM := false
+			headerLineNum := 0
+			firstLineIsCurlyBrace := false
+
+			for scanner.Scan() {
+				v := scanner.Text()
+
+				if !checkBOM {
+					v = strings.TrimPrefix(v, utf8BOM)
+					checkBOM = true
+				}
+
+				if headerLineNum <= 1 {
+					cleanedLine := strings.Replace(strings.Replace(v, " ", "", -1), "\t", "", -1)
+					cleanedLine = strings.TrimSuffix(cleanedLine, "[") // don't care which line this is on, could be third line
+					if cleanedLine == "{" && headerLineNum == 0 {
+						firstLineIsCurlyBrace = true
+					} else {
+						const jsonStart = "{\"Files\":"
+						jsonStartNoBrace := strings.TrimPrefix(jsonStart, "{")
+						isJSON := cleanedLine == jsonStart || firstLineIsCurlyBrace && cleanedLine == jsonStartNoBrace
+						if isJSON {
+							glcm.Error("The format for list-of-files has changed. The old JSON format is no longer supported")
+						}
+					}
+					headerLineNum++
+				}
```

Comment:
this entire check as doesn't apply to versions. This was checking whether the old list-of-files format (JSON) is mistakenly used by the user.

Reply:
Yes. I was thinking the same. I wanted to confirm with you once.

Reply:
Done